lib/relations_restful/__init__.py

```diff
@@ -1,6 +1,43 @@
 """
-Unittest Tools for Relations
+Utilities for Relations RESTful
 """
 
-from relations_restful.resource import Resource, exceptions
+import inspect
+
+from relations_restful.resource import ResourceIdentity, Resource, exceptions
 from relations_restful.source import Source
+
+def resources(module):
+    """
+    List all the resources in a module
+    """
+
+    return [
+        m[1]
+        for m in inspect.getmembers(
+            module,
+            lambda model: inspect.isclass(model)
+            and issubclass(model, Resource)
+        )
+    ]
+
+def ensure(module, models):
+    """
+    Creates the Resources for all models
+    """
+
+    exists = [resource.MODEL for resource in resources(module)]
+
+    return [
+        type(model.__name__, (Resource, ), {'MODEL': model})
+        for model in models if model not in exists
+    ]
+
+def attach(restful, module, models):
+    """
+    Attach all Reources to a Restful
+    """
+
+    for resource in resources(module) + ensure(module, models):
+        if resource.__name__.lower() not in restful.endpoints:
```

Comment:
Should this throw a warning if `resource.__name__.lower() not in restful.endpoints` evaluates to False (meaning the endpoint was already there). I understand this is there to handle the case where the dev already attached the route or otherwise customized it.. but there is a hidden gotcha if you unintentionally add a different endpoint that happens to have a colliding name. This would silently fail in that situation.